code-journal/src/FormElement.tsx

```diff
@@ -0,0 +1,183 @@
+import { FormEvent, useEffect, useRef, useState } from 'react';
+import {
+  addEntry,
+  Entry,
+  readEntry,
+  removeEntry,
+  updateEntry,
+} from './data.ts';
+import { useNavigate, useParams } from 'react-router-dom';
+import { createPortal } from 'react-dom';
+const imgSrc = '/placeholder-image-square.jpg';
+
+export function FormElement() {
+  const [newTitle, setNewTitle] = useState<string>();
+  const [newImg, setNewImg] = useState<string>();
+  const [newNote, setNewNote] = useState<string>();
+  const [entryEdit, setEntryEdit] = useState<Entry>();
+  const [deleteConfirmDisplayed, setDeleteConfirmDisplayed] = useState(false);
+  const { id } = useParams();
+  const imgRef = useRef<HTMLImageElement>(null);
+  const navigate = useNavigate();
+  function handleEntrySubmission(e: FormEvent<HTMLFormElement>) {
+    e.preventDefault();
+    if (!entryEdit) {
+      addEntry({
+        title: newTitle ?? '',
+        notes: newNote ?? '',
+        photoUrl: newImg ?? '',
+      });
+    } else {
+      updateEntry({
+        title: newTitle ?? '',
+        notes: newNote ?? '',
+        photoUrl: newImg ?? '',
+        entryId: entryEdit.entryId,
+      });
+    }
+    navigate('/entries');
+  }
+  useEffect(() => {
+    const changeImage = async function (): Promise<void> {
+      try {
+        const fgh = await new Promise((resolve, reject) => {
+          if (!(imgRef.current instanceof HTMLImageElement))
+            throw new Error('imgRef not assigned to img');
+          imgRef.current.onload = () => resolve('img url was valid');
+          imgRef.current.onerror = () => reject(new Error('url not valid'));
+        });
+        console.log(fgh);
```

Comment:
Since you registered onload and onerror handlers, you need to remove them in a useEffect clean-up. So not only has ChatGPT given you ugly, useless code, but it is worse than if you had done nothing since you now have a memory leak.